Applications/Utils/FileConverter/NetCdfConverter.cpp

```diff
@@ -0,0 +1,696 @@
+/**
+ * @copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+// STL
+#include <cctype>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include <tclap/CmdLine.h>
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+
+#include <netcdf>
+
+// BaseLib
+#include "BaseLib/FileTools.h"
+#include "BaseLib/LogogSimpleFormatter.h"
+#include "InfoLib/GitInfo.h"
+
+// GeoLib
+#include "GeoLib/Raster.h"
+
+// MeshLib
+#include "MeshLib/Mesh.h"
+#include "MeshLib/MeshGenerators/RasterToMesh.h"
+
+#include "MeshLib/IO/VtkIO/VtuInterface.h"
+
+using namespace netCDF;
+
+void checkExit(std::string const& str)
+{
+    if (str == "x" || str == "exit")
+        exit(0);
+}
+
+void showErrorMessage(std::size_t error_id, std::size_t max = 0)
+{
+    if (error_id == 0)
+    {
+        ERR("Input not valid.");
+    }
+    else if (error_id == 1)
+    {
+        ERR("Index not valid. Valid indices are in [0,%d].", max);
+    }
+    else if (error_id == 2)
+    {
+        showErrorMessage(0);
+        std::cout << "Type \"info\" to display the available options again. "
+                     "\"exit\" will exit the programme.\n";
+    }
+}
+
+std::size_t parseInput(std::string const& request_str, std::size_t max_val,
+                       bool has_info = false)
+{
+    while (true)
+    {
+        std::cout << request_str;
+        std::string str;
+        std::size_t val;
+        std::getline(std::cin, str);
+        checkExit(str);
+        if (has_info && str == "info")
+            return (max_val);
+        std::stringstream str_stream(str);
+        if (!(str_stream >> val))
+        {
+            std::size_t const error_val = (has_info) ? 2 : 0;
+            showErrorMessage(error_val);
+            continue;
+        }
+        if (val > max_val - 1)
+        {
+            showErrorMessage(1, max_val - 1);
+            continue;
+        }
+        return val;
+    }
+    return std::numeric_limits<std::size_t>::max();
+}
+
+NcVar getDimVar(NcFile const& dataset, NcVar const& var, std::size_t const dim)
+{
+    NcDim const& dim_obj = var.getDim(dim);
+    return dataset.getVar(dim_obj.getName());
+}
+
+std::vector<std::string> showArrays(NcFile const& dataset)
+{
+    std::size_t const n_vars(dataset.getDimCount());
+    std::cout << "The NetCDF file contains the following " << n_vars
+              << " arrays:\n\n";
+    std::cout << "\tIndex\tArray Name\t#Dimensions\n";
+    std::cout << "-------------------------------------------\n";
+    auto const& names = dataset.getVars();
+    std::vector<std::string> var_names;
```

Comment:
I don't know what you're trying to say. Can you explain please?

Reply:
One function does the "push_back" to create a vector, another prints that vector. This is what I mean by separation.

Reply:
I've added the change ... can you please [take a look?](https://github.com/ufz/ogs/pull/2654/commits/c742cb45f84aed9d47e87225bfadd7ff4926fdba)

In my opinion it doesn't make sense to do this because I only need the variable name vector for the selection. I now have two almost identical functions and need to parse the file two times instead of once, only to get a vector than I don't need after the user selection.

Reply:
Looks fine to me.

Reply:
:heavy_check_mark: